sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -693,6 +719,12 @@ public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
 
     @Override
     public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offersToProcess) {
+        if (!apiServerDisabled.get() && (apiServer == null || !apiServer.isStarted())) {
+            LOGGER.info("Waiting for API Server to start ...");
+            declineOffers(driver, Collections.emptyList(), offersToProcess);
+            return;
```

Comment:
I'd be interested to see how that affects Elastic, since it takes 1.5-2 minutes for its API server to come up. Would that make a scheduler restart part of the normal process of installing the package? https://jira.mesosphere.com/browse/INFINITY-996